src/Neo.CLI/CLI/MainService.Tools.cs

```diff
@@ -426,88 +417,60 @@ public string Base64Fixed(string str)
         [ParseFunction("Base64 Smart Contract Script Analysis")]
         private string? ScriptsToOpCode(string base64)
         {
-            List<byte> scripts;
-            try
-            {
-                scripts = Convert.FromBase64String(base64).ToList();
-            }
-            catch (Exception)
-            {
-                return null;
-            }
+            Script script;
             try
             {
-                _ = new Script(scripts.ToArray(), true);
+                var scriptData = Convert.FromBase64String(base64);
+                script = new Script(scriptData.ToArray(), true);
             }
             catch (Exception)
             {
                 return null;
             }
-            return ScriptsToOpCode(scripts);
+            return ScriptsToOpCode(script);
         }
 
-        private string ScriptsToOpCode(List<byte> scripts)
+        private string ScriptsToOpCode(Script script)
         {
-            //Initialize all OpCodes
-            var OperandSizePrefixTable = new int[256];
-            var OperandSizeTable = new int[256];
-            foreach (FieldInfo field in typeof(OpCode).GetFields(BindingFlags.Public | BindingFlags.Static))
-            {
-                var attribute = field.GetCustomAttribute<OperandSizeAttribute>();
-                if (attribute == null) continue;
-                int index = (int)(OpCode)field.GetValue(null);
-                OperandSizePrefixTable[index] = attribute.SizePrefix;
-                OperandSizeTable[index] = attribute.Size;
-            }
             //Initialize all InteropService
             var dic = new Dictionary<uint, string>();
             ApplicationEngine.Services.ToList().ForEach(p => dic.Add(p.Value.Hash, p.Value.Name));
 
             //Analyzing Scripts
+            var ip = 0;
+            Instruction instruction;
             var result = new List<string>();
-            while (scripts.Count > 0)
+            while ((instruction = script.GetInstruction(ip)) != null)
```

Comment:
@chenzhitong could you test it?

Reply:
**Test failed.**

DARkYXRhAgBlzR0MFPdcrAXPVptVduMEs2lf1jQjxKIKDBT3XKwFz1abVXbjBLNpX9Y0I8SiChTAHwwIdHJhbnNmZXIMFKNSbimM12LkFYX/8KGvm2ttFxulQWJ9W1I=

I was expecting it to output. 
```
PUSHDATA1 data
PUSHINT32 500000000
PUSHDATA1 0x0aa2c42334d65f69b304e376559b56cf05ac5cf7
PUSHDATA1 0x0aa2c42334d65f69b304e376559b56cf05ac5cf7
PUSH4
PACK
PUSH15
PUSHDATA1 transfer
PUSHDATA1 0xa51b176d6b9bafa1f0ff8515e462d78c296e52a3
SYSCALL System.Contract.Call
```

Now it's outputting.
```
PUSHDATA1 data
PUSHINT32 500000000
PUSHDATA1 0x0aa2c42334d65f69b304e376559b56cf05ac5cf7
PUSHDATA1 0x0aa2c42334d65f69b304e376559b56cf05ac5cf7
PUSH4
PACK
PUSH15
PUSHDATA1 transfer
PUSHDATA1 0xa51b176d6b9bafa1f0ff8515e462d78c296e52a3
```

DCECbzTesnBofh/Xng1SofChKkBC7jhVmLxCN1vk\u002B49xa2pBVuezJw==

I was expecting it to output. 
```
PUSHDATA1 026f34deb270687e1fd79e0d52a1f0a12a4042ee385598bc42375be4fb8f716b6a
SYSCALL System.Crypto.CheckSig
```

Now it's outputting.
```
PUSHDATA1 026f34deb270687e1fd79e0d52a1f0a12a4042ee385598bc42375be4fb8f716b6a
```
@shargon

Reply:
I fixed it